src/metpy/calc/tools.py

```diff
@@ -787,35 +800,48 @@ def lat_lon_grid_deltas(longitude, latitude, **kwargs):
         longitude = np.asarray(longitude)
         latitude = np.asarray(latitude)
 
+    # Determine dimension order for offset slicing
+    take_y = make_take(latitude.ndim, kwargs.pop('y_dim', -2))
```

Comment:
Think this and `x_dim` could become a kwarg-only argument?

Reply:
Sure. Looking back at this, I'm not sure why I did it the way it is now.